src/components/gauge/gauge.tsx

```diff
@@ -0,0 +1,445 @@
+   * Define the color of the center in _(Circle)_ gauge type.
+   * Value in *hex, rgb, rgba, hsl, cmyk* format or *color name*.
+   * _(Transparent by Default)_
+   */
+  @Prop() styleCenterColor = "transparent";
+
+  /**
+   * Property of type Style
+   * Define the color of the center text in _(Circle)_ gauge type.
+   * Value in *hex, rgb, rgba, hsl, cmyk* format or *color name*.
+   * _(Gray by Default)_
+   */
+  @Prop() styleCenterTextColor = "gray";
+
+  /**
+   * Property of type Style.
+   * Define if shadow will display or not. Default is disabled.
+   */
+  @Prop() styleShadow = false;
+
+  /**
+   * Property of type Style.
+   * Define if border will display or not. Default is disabled.
+   */
+  @Prop() styleBorder = false;
+
+  /**
+   * Property of type Style.
+   * Define the border width. Value in *px*.
+   */
+  @Prop() styleBorderWidth = 2;
+
+  /**
+   * Property of type Style.
+   * Define the border color. Value in *hex, rgb, rgba, hsl, cmyk* format or *color name*.
+   */
+  @Prop() styleBorderColor = "rgb(0, 92, 129)";
+
+  /**
+   * This property allows you to select the gauge type. _(Circle or Line)_.
+   * Default is linear type.
+   */
+  @Prop() gaugeType: "Line" | "Circle" = "Line";
+
+  /**
+   *  Allows display current value. Default is disabled.
+   *
+   */
+  @Prop() showValue = false;
+
+  /**
+   * The Minimum Value of the gauge
+   *
+   */
+  @Prop() minValue: number;
+
+  /**
+   * The Currecnt value indicanding in the gauge
+   *
+   */
+  @Prop() currentValue: number = this.minValue;
+
+  /**
+   * This allows specifying the width of the circumference _(When gauge is Circle type)_ and the width of the bar _(When gauge is Line type)_ in % relative the component size.
+   *
+   */
+  @Prop() thickness = 10;
+
+  @State() maxValue: number;
+
+  @State() totValues = this.minValue;
+
+  @State() children = [];
+
+  @State() minorSize: number;
+
+  @Listen("gxGaugeRangeDidLoad")
+  onGaugeRangeDidLoad({ detail: childRange }) {
+    this.children = [...this.children, childRange];
+    this.totValues += childRange.amount;
+    childRange.element.addEventListener("gxGaugeRangeDidUnload", () => {
+      const inDex = this.children.findIndex(x => x === childRange);
+      this.children.splice(inDex, 1);
+      this.totValues -= childRange.amount;
+    });
+    childRange.element.addEventListener("gxGaugeRangeDidUpdate", () => {
+      const inDex = this.children.findIndex(x => x === childRange);
+      this.children.splice(inDex, 1, childRange);
+      this.totValues = 0;
+      for (const childInstance of this.children) {
+        this.totValues += childInstance.amount;
+      }
+    });
+  }
+
+  componentWillLoad() {
+    this.totValues = 0;
+  }
+
+  private calcThickness() {
+    return typeof this.thickness === "number" &&
+      (this.thickness > 0 && this.thickness <= 100)
+      ? this.thickness / 5
+      : 10;
+  }
+
+  ////////////////////////////////////////
+
+  render() {
```

Comment:
Please take a second look at the `render` method. It got too big. Maybe it would be a good idea to split it in a `renderCircle` and a `renderLine` method.